ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt

```diff
@@ -217,28 +219,40 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
 
   @TestOnly fun getButtonsIndex() = buttonsIndex
 
-  private fun showLoiNameDialog(initialTextValue: String) {
-    (view as ViewGroup).addView(
-      ComposeView(requireContext()).apply {
-        setContent {
-          val openAlertDialog = remember { mutableStateOf(Pair(initialTextValue, true)) }
-          when {
-            openAlertDialog.value.second -> {
-              val (textFieldValue, openState) = openAlertDialog.value
-              LoiNameDialog(
-                textFieldValue = textFieldValue,
-                onConfirmRequest = {
-                  openAlertDialog.value = Pair(textFieldValue, false)
-                  handleLoiNameSet(loiName = textFieldValue)
-                },
-                onDismissRequest = { openAlertDialog.value = Pair(initialTextValue, false) },
-                onTextFieldChange = { openAlertDialog.value = Pair(it, openState) }
-              )
+  private fun launchLoiNameDialog() {
+    dataCollectionViewModel.loiNameDialogOpen.value = true
+    lifecycleScope.launch {
+      (view as ViewGroup).addView(
+        ComposeView(requireContext()).apply {
+          setContent {
+            // The LOI NameDialog should call `handleLoiNameSet()` to continue to the next task.
+            ShowLoiNameDialog(dataCollectionViewModel.loiName.value ?: "") {
+              handleLoiNameSet(loiName = it)
             }
           }
         }
-      }
-    )
+      )
+    }
+  }
+
+  @Composable
+  fun ShowLoiNameDialog(initialNameValue: String, onNameSet: (String) -> Unit) {
```

Comment:
Pity we don't have a `GeometryTaskFragment`, since `AbstractTaskFragment` is used by all tasks.